src/target/riscv/riscv.c

```diff
@@ -3001,6 +3001,54 @@ bool riscv_has_register(struct target *target, int hartid, int regid)
 	return 1;
 }
 
+/**
+ * If write is true:
+ *   return true iff we are guaranteed that the register will contain exactly
+ *       the value we just wrote when it's read.
+ * If write is false:
+ *   return true iff we are guaranteed that the register will read the same
+ *       value in the future as the value we just read.
+ */
+static bool gdb_regno_cacheable(enum gdb_regno regno, bool write)
+{
+	/* GPRs, FPRs, vector registers are just normal data stores. */
+	if (regno <= GDB_REGNO_XPR31 ||
+			(regno >= GDB_REGNO_FPR0 && regno <= GDB_REGNO_FPR31) ||
+			(regno >= GDB_REGNO_V0 && regno <= GDB_REGNO_V31) ||
+			regno == GDB_REGNO_PC)
+		return true;
+
+	/* Most CSRs won't change value on us, but we can't assume it about rbitrary
```

Comment:
typo?  ("arbitrary"?)

Reply:
I'll fix this some other time, in the interest of merging this ASAP for Kevin.